### DISCUSSION TITLE
请问游戏剧情存储问题？

### DISCUSSION THREAD

**Discussion by rocketman123456** [BoomingTech/Piccolo#234] [Q&A] [answered]
想请问一下，像是底特律变人这样的重剧情游戏，一般会采用什么方法去存储剧情数据呢？或者是那种文字冒险游戏，是有什么通用的方法组织管理这些数据呢？

**Comment by hyv1001** [comment, marked as answer]
剧情数据一般包括过场动画（一般通过Sequencer制作），剧情分支处理（一般通过行为树）和很多配套的音视频文件。
一般这些数据都是游戏Asset资源，所以数据存储上和普通的游戏资产一样存储在磁盘上，由引擎内的文件系统进行快速读取，触发相关的游戏逻辑。
网游和单机有很些不同，网游中一些任务相关的剧情数据是存储在服务器数据库中。
另外，现在也有一些游戏此类数据是存储在云端的，可以边玩边下载到本地。
